Unable to test canvas in headless build

## Expected Behavior
Right now there is a canvas test that should pass CI

## Current Behavior
It fails due to being ran in a headless instance without an X instances.

## Steps to Reproduce (for bugs)
1. Run the tests on a server

## Context
Trying to release on AppCenter. Houston runs the build and tests on a server without X.

## Screenshots of the Problem

I'll do you one better and just copy the log:
```
[152/154] cc -Itests/59830eb@@canvas_tests@exe -Itests -I../../tests -I/build/com.github.akiraux.akira-0.0.1/tests/../src -I/build/com.github.akiraux.akira-0.0.1/debian/build/tests/../src -Isrc -I../../src -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib/x86_64-linux-gnu/dbus-1.0/include -I/usr/include/gtk-3.0 -I/usr/include/gio-unix-2.0/ -I/usr/include/harfbuzz -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/include/granite -I/usr/include/gee-0.8 -I/usr/include/libxml2 -I/usr/include/goocanvas-2.0 -I/usr/include/json-glib-1.0 -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -w -g '-DGETTEXT_PACKAGE="com.github.akiraux.akira"' '-DG_LOG_DOMAIN="Akira"' -g -O2 -fdebug-prefix-map=/build/com.github.akiraux.akira-0.0.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -pthread '-DTEST_DATA_DIR="/build/com.github.akiraux.akira-0.0.1/tests/data/schemas"'  -MD -MQ 'tests/59830eb@@canvas_tests@exe/meson-generated_canvas_test-fills-item.c.o' -MF 'tests/59830eb@@canvas_tests@exe/meson-generated_canvas_test-fills-item.c.o.d' -o 'tests/59830eb@@canvas_tests@exe/meson-generated_canvas_test-fills-item.c.o' -c 'tests/59830eb@@canvas_tests@exe/canvas/test-fills-item.c'
[153/154] cc -Itests/59830eb@@canvas_tests@exe -Itests -I../../tests -I/build/com.github.akiraux.akira-0.0.1/tests/../src -I/build/com.github.akiraux.akira-0.0.1/debian/build/tests/../src -Isrc -I../../src -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib/x86_64-linux-gnu/dbus-1.0/include -I/usr/include/gtk-3.0 -I/usr/include/gio-unix-2.0/ -I/usr/include/harfbuzz -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/include/granite -I/usr/include/gee-0.8 -I/usr/include/libxml2 -I/usr/include/goocanvas-2.0 -I/usr/include/json-glib-1.0 -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -w -g '-DGETTEXT_PACKAGE="com.github.akiraux.akira"' '-DG_LOG_DOMAIN="Akira"' -g -O2 -fdebug-prefix-map=/build/com.github.akiraux.akira-0.0.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -pthread '-DTEST_DATA_DIR="/build/com.github.akiraux.akira-0.0.1/tests/data/schemas"'  -MD -MQ 'tests/59830eb@@canvas_tests@exe/meson-generated_tests-model.c.o' -MF 'tests/59830eb@@canvas_tests@exe/meson-generated_tests-model.c.o.d' -o 'tests/59830eb@@canvas_tests@exe/meson-generated_tests-model.c.o' -c 'tests/59830eb@@canvas_tests@exe/tests-model.c'
[154/154] cc  -o tests/canvas_tests 'tests/59830eb@@canvas_tests@exe/meson-generated_tests-model.c.o' 'tests/59830eb@@canvas_tests@exe/meson-generated_canvas_test-fills-item.c.o' 'tests/59830eb@@canvas_tests@exe/meson-generated_canvas_test-main.c.o' -Wl,--no-undefined -Wl,--as-needed -g -O2 -fdebug-prefix-map=/build/com.github.akiraux.akira-0.0.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,--start-group src/libakira-library-1.0.so /usr/lib/x86_64-linux-gnu/libgtk-3.so /usr/lib/x86_64-linux-gnu/libgdk-3.so /usr/lib/x86_64-linux-gnu/libpangocairo-1.0.so /usr/lib/x86_64-linux-gnu/libpango-1.0.so /usr/lib/x86_64-linux-gnu/libatk-1.0.so /usr/lib/x86_64-linux-gnu/libcairo-gobject.so /usr/lib/x86_64-linux-gnu/libcairo.so /usr/lib/x86_64-linux-gnu/libgdk_pixbuf-2.0.so /usr/lib/x86_64-linux-gnu/libgio-2.0.so /usr/lib/x86_64-linux-gnu/libgobject-2.0.so /usr/lib/x86_64-linux-gnu/libglib-2.0.so /usr/lib/x86_64-linux-gnu/libgranite.so /usr/lib/x86_64-linux-gnu/libgee-0.8.so /usr/lib/x86_64-linux-gnu/libxml2.so /usr/lib/x86_64-linux-gnu/libgoocanvas-2.0.so /usr/lib/x86_64-linux-gnu/libarchive.so /usr/lib/x86_64-linux-gnu/libjson-glib-1.0.so -lm -Wl,--end-group '-Wl,-rpath,$ORIGIN/../src' -Wl,-rpath-link,/build/com.github.akiraux.akira-0.0.1/debian/build/src  
make[1]: Leaving directory '/build/com.github.akiraux.akira-0.0.1'
   debian/rules override_dh_auto_test
make[1]: Entering directory '/build/com.github.akiraux.akira-0.0.1'
cd debian/build && ninja test
[0/1] Running all tests.
1/1 canvas_tests                            FAIL     0.04 s (exit status 1)

Ok:                    0
Expected Fail:         0
Fail:                  1
Unexpected Pass:       0
Skipped:               0
Timeout:               0


The output from the failed tests:

1/1 canvas_tests                            FAIL     0.04 s (exit status 1)

--- command ---
G_TEST_SRCDIR='/build/com.github.akiraux.akira-0.0.1/tests' GSETTINGS_SCHEMA_DIR='/build/com.github.akiraux.akira-0.0.1/debian/build/data/schemas' GSETTINGS_BACKEND='memory' G_TEST_BUILDDIR='/build/com.github.akiraux.akira-0.0.1/debian/build/tests' /build/com.github.akiraux.akira-0.0.1/debian/build/tests/canvas_tests
--- stderr ---
Unable to init server: Could not connect: Connection refused

(canvas_tests:12427): Gtk-WARNING **: 00:02:26.372: cannot open display: 
-------

Full log written to /build/com.github.akiraux.akira-0.0.1/debian/build/meson-logs/testlog.txt
FAILED: meson-test 
/usr/bin/meson test --no-rebuild --print-errorlogs
ninja: build stopped: subcommand failed.
debian/rules:26: recipe for target 'override_dh_auto_test' failed
make[1]: *** [override_dh_auto_test] Error 1
make[1]: Leaving directory '/build/com.github.akiraux.akira-0.0.1'
debian/rules:13: recipe for target 'build' failed
make: *** [build] Error 2
dpkg-buildpackage: error: debian/rules build subprocess returned exit status 2
E: Failed autobuilding of package
I: unmounting dev/ptmx filesystem
I: unmounting dev/pts filesystem
I: unmounting dev/shm filesystem
I: unmounting proc filesystem
I: unmounting sys filesystem
I: cleaning the build env 
I: removing directory /var/cache/liftoff-build//7664 and its subdirectories
```

## Your Environment (for bugs)
Linux echo 4.4.0-177-generic #207-Ubuntu SMP Mon Mar 16 01:16:10 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
```
NAME="Ubuntu"
VERSION="16.04.6 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.6 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
VERSION_CODENAME=xenial
UBUNTU_CODENAME=xenial
```
Probably testable with houston ci docker container

- Akira Version: v0.0.1 tag

--- a/meson.build
+++ b/meson.build
@@ -4,6 +4,7 @@ project('com.github.akiraux.akira', 'vala', 'c')
 akira_prefix = get_option('prefix')
 akira_datadir = join_paths(akira_prefix, get_option('datadir'))
 akira_pkgdatadir = join_paths(akira_datadir, meson.project_name())
+install_tests = get_option('install-tests')
 
 vala_version = meson.get_compiler('vala').version()
 if vala_version.version_compare('<=0.43.90')
@@ -67,6 +68,9 @@ add_global_arguments(['-DG_LOG_DOMAIN="Akira"'], language:'c')
 subdir('src')
 subdir('data')
 subdir('po')
-subdir('tests')
+
+if install_tests
+    subdir('tests')
+endif
 
 meson.add_install_script('build-aux/meson/post_install.py')

--- a/meson_options.txt
+++ b/meson_options.txt
@@ -1,9 +1,4 @@
-option(
-        'profile',
-        type: 'combo',
-        choices: [
-                'default',
-                'development'
-        ],
-        value: 'default'
-)
+option('profile', type: 'combo', choices: ['default', 'development'], value: 'default')
+
+option('install-tests', type : 'boolean', value : 'false',
+       description : 'install test executables')

